tests/autoyast/repos.pm

```diff
@@ -30,7 +30,11 @@ use utils;
 sub run {
     # sles12_minimal.xml profile does not install "ip"
     assert_script_run 'ip a || ifstatus all';
-    quit_packagekit;
+    if (!check_var('DESKTOP', 'textmode')) {
+         quit_packagekit;
+         # poo#87850 wait for packagekit related zypper process to exit
+         assert_script_run 'until ! pgrep zypper; do sleep 1; done';
```

Comment:
1) if packagekit call zypper but can not stop zypper after systemctl stop, is it an issue?
2) i suggest move the logic directly into function quit_packagekit 
3) no idea why we need check DESKTOP, packagekit only support gnome?

Reply:
The function quit_packagekit will quit the service, these related zypper process will quit also but need wait.
I think it is not very suitable to move the wait logic to quit_packagekit, because the purpose of it just to ensure the packagekit quit. Anyway, I will ask owner to review this point also.
It should say that only gnome support the packagekit.

Reply:
I agree it should be inside that function. I also was not aware of any packagekit related zypper process, stopping packagekit has been there for a while, strange that now needs this extra step.

Reply:
Ok, so I will move the logic to quit_packagekit. About the packagekit related zypper process, I just get this from actual test jobs for migration test and suggestion from Tony.

Reply:
If you only have these problems in specific cases I suggest to keep it here or for the special cases but not do in general within `quit_packagekit`. This might even be a product bug that quitting packagekit does not actually stop all related process which it likely should do.

Reply:
right, if you just got that from result of test then is better as @okurz mentioned, beside that function is used everywhere, so better keep it isolated this change and even file a file for that to get right expectations.

Reply:
Yes, this only happened in specific cases. So I will only modify the code of repos here. Thanks Oliver and jknphy.